enterprise/utils.py

```diff
@@ -166,3 +173,156 @@ def get_catalog_admin_url_template():
 
     # Return matched FQDN from catalog api url appended with catalog admin path
     return match.group("fqdn").rstrip("/") + "/admin/catalogs/catalog/{catalog_id}/change/"
+
+
+def build_notification_message(template_context, template_configuration=None):
+    """
+    Create HTML and plaintext message bodies for a notification.
+
+    We receive a context with data we can use to render, as well as an optional site
+    template configration - if we don't get a template configuration, we'll use the
+    standard, built-in template.
+
+    Arguments:
+        template_context (dict): A set of data to render
+        template_configuration: A database-backed object with templates
+            stored that can be used to render a notification.
+    """
+    if (
+            template_configuration is not None and
+            template_configuration.html_template and
+            template_configuration.plaintext_template
+    ):
+        plain_msg, html_msg = template_configuration.render_all_templates(template_context)
+    else:
+        plain_msg = render_to_string(
+            'enterprise/emails/user_notification.txt',
+            template_context
+        )
+        html_msg = render_to_string(
+            'enterprise/emails/user_notification.html',
+            template_context
+        )
+
+    return plain_msg, html_msg
+
+
+def get_notification_subject_line(course_name, template_configuration=None):
+    """
+    Get a subject line for a notification email.
+
+    The method is designed to fail in a "smart" way; if we can't render a
+    database-backed subject line template, then we'll fall back to a template
+    saved in the Django settings; if we can't render _that_ one, then we'll
+    fall through to a friendly string written into the code.
+
+    Arguments:
+        course_name (str): Course name to be rendered into the string
+        template_configuration: A database-backed object with a stored subject line template
+    """
+    stock_subject_template = _('You\'ve been enrolled in {course_name}!')
+    default_subject_template = getattr(
+        settings,
+        'ENTERPRISE_ENROLLMENT_EMAIL_DEFAULT_SUBJECT_LINE',
+        stock_subject_template,
+    )
+    if template_configuration is not None and template_configuration.subject_line:
+        final_subject_template = template_configuration.subject_line
+    else:
+        final_subject_template = default_subject_template
+
+    try:
+        return final_subject_template.format(course_name=course_name)
+    except KeyError:
+        pass
+
+    try:
+        return default_subject_template.format(course_name=course_name)
+    except KeyError:
+        return stock_subject_template.format(course_name=course_name)
+
+
+def send_notification_message(user, enrolled_in, enterprise_customer, email_connection=None):
```

Comment:
If this method is solely-intended to handle email communications we should probably be more specific with the name -- eg `send_email_notification_message`